roles/idrac_gather_facts/tasks/main.yml

```diff
@@ -1,162 +1,103 @@
-  when: target is defined and "Fan" in target
-
-- name: Gather Firmware facts.
-  ansible.builtin.include_tasks: get_firmware_info.yml
-  when: target is defined and "Firmware" in target
-
-- name: Gather HostNIC facts.
-  ansible.builtin.include_tasks: get_host_nic_info.yml
-  when: target is defined and "HostNIC" in target
-
-- name: Gather License facts.
-  ansible.builtin.include_tasks: get_license_info.yml
-  when: target is defined and "License" in target
-
-- name: Gather Memory facts.
-  ansible.builtin.include_tasks: get_memory_info.yml
-  when: target is defined and "Memory" in target
-
-- name: Gather NIC facts.
-  ansible.builtin.include_tasks: get_nic_info.yml
-  when: target is defined and "NIC" in target
-
-- name: Gather PCIeSSDBackPlane facts.
-  ansible.builtin.include_tasks: get_backplane_info.yml
-  when: target is defined and "PCIeSSDBackPlane" in target
-
-- name: Gather PowerSupply facts.
-  ansible.builtin.include_tasks: get_power_supply_info.yml
-  when: target is defined and "PowerSupply" in target
-
-- name: Gather PresenceAndStatusSensor facts.
-  ansible.builtin.include_tasks: get_pas_sensor_info.yml
-  when: target is defined and "PresenceAndStatusSensor" in target
-
-- name: Gather Sensors Battery facts.
-  ansible.builtin.include_tasks: get_battery_info.yml
-  when: target is defined and "Sensors_Battery" in target
-
-- name: Gather Sensors Intrusion facts.
-  ansible.builtin.include_tasks: get_intrusion_info.yml
-  when: target is defined and "Sensors_Intrusion" in target
-
-- name: Gather Sensors Voltage facts.
-  ansible.builtin.include_tasks: get_voltage_info.yml
-  when: target is defined and "Sensors_Voltage" in target
-
-- name: Gather VirtualDisk facts.
-  ansible.builtin.include_tasks: get_virtual_disk_info.yml
-  when: target is defined and "VirtualDisk" in target
-
-- name: Gather PCIeDevice facts.
-  ansible.builtin.include_tasks: get_pcie_device_info.yml
-  when: target is defined and "PCIeDevice" in target
-
-- name: Gather PhysicalDisk facts.
-  ansible.builtin.include_tasks: get_physical_info.yml
-  when: target is defined and "PhysicalDisk" in target
-
-- name: Gather SystemMetrics facts.
-  ansible.builtin.include_tasks: get_metrics_info.yml
-  when: target is defined and "SystemMetrics" in target
+- name: IDRAC Gather facts
+  module_defaults:
+    ansible.builtin.uri:
+      validate_certs: "{{ validate_certs }}"
+      ca_path: "{{ ca_path | default(omit) }}"
+      method: "{{ idrac_gather_facts_uri_method }}"
+      user: "{{ username | default(lookup('env', 'IDRAC_USERNAME')) }}"
+      password: "{{ password | default(lookup('env', 'IDRAC_PASSWORD')) }}"
+      timeout: "{{ https_timeout }}"
+      force_basic_auth: true
+      headers: "{{ idrac_gather_facts_uri_headers }}"
+      body_format: "{{ idrac_gather_facts_uri_body_format }}"
+      status_code: "{{ idrac_gather_facts_uri_status_code }}"
+      return_content: "{{ idrac_gather_facts_uri_return_content }}"
+  vars:
```

Comment:
couldn't we move this to vars.yml ?

Reply:
module_defaults is better, as will apply to uri module only